src/Parser/Warlock/Demonology/Modules/Features/SpellUsable.js

```diff
@@ -0,0 +1,45 @@
+import SPELLS from 'common/SPELLS';
+import CoreSpellUsable from 'Parser/Core/Modules/SpellUsable';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import GlobalCooldown from 'Parser/Core/Modules/GlobalCooldown';
+import T20_2set from 'Parser/Warlock/Demonology/Modules/Items/T20_2set';
+
+
+class SpellUsable extends CoreSpellUsable{
+  lastPotentialTrigger= null;
+
+  static dependencies = {
+    ...CoreSpellUsable.dependencies,
+    combatants: Combatants,
+    globalCooldown: GlobalCooldown,
+    t20_2set: T20_2set,
+  };
+
+  on_initialized(){
+    this.has2set = this.combatants.selected.hasBuff(SPELLS.WARLOCK_DEMO_T20_2P_BONUS.id);
+  }
+
+  on_byPlayer_cast(event){
+    if(super.on_byPlayer_cast){
+      super.on_byPlayer_cast(event);
+    }
+
+    const spellId = event.ability.guid;
+    if((spellId === SPELLS.DEMONBOLT_TALENT.id || spellId === SPELLS.SHADOW_BOLT.id || spellId === SPELLS.DEMONWRATH_CAST.id) && this.has2set){
+        this.lastPotentialTrigger = event.timestamp;
+    } else if(spellId === SPELLS.CALL_DREADSTALKERS.id){
+      this.lastPotentialTrigger = null;
+    }
+  }
+
+  beginCooldown(spellId, timestamp){
+    if(spellId === SPELLS.CALL_DREADSTALKERS.id){
```

Comment:
You can also join the conditions together to avoid nesting (because of short circuit evaluation, it shouldn't check cooldown of anything else than Dreadstalkers, but correct me if I'm wrong 😄 ):
```javascript
if (spellId === SPELLS.CALL_DREADSTALKERS.id && this.isOnCooldown(spellId)) {
... 
}
```